rio/src/RIO/Process.hs

```diff
@@ -582,12 +602,15 @@ findExecutable name = do
             let loop [] = return $ Left $ ExecutableNotFound name (pcPath pc)
                 loop (dir:dirs) = do
                     let fp0 = dir FP.</> name
-                        fps0 = map (fp0 ++) (pcExeExtensions pc)
+                               -- On Windows, include the name if it has an
+                               -- existing extension
+                        fps0 = (if isWindows && FP.hasExtension fp0
+                                  then (fp0:) else id)
+                               (map (fp0 ++) (pcExeExtensions pc))
                         testFPs [] = loop dirs
                         testFPs (fp:fps) = do
                             exists <- D.doesFileExist fp
-                            existsExec <- if exists then liftM D.executable $ D.getPermissions fp else return False
```

Comment:
This looks broken: it isn't checking for the executable bit on non-Windows platforms.

Reply:
You are correct. In the original 'non-Windows' code there was an inconsistency in the two limbs. My version had no `isExecutable` filter in both limbs. I've now added the `isExecutable` filter to both limbs in a revised version. I need, however, to do some testing - it will be a little while until I have access to a non-Windows machine to do that.